apps/prairielearn/src/pages/instructorAssessmentQuestions/components/InstructorAssessmentQuestionsTable.tsx

```diff
@@ -47,81 +58,915 @@ function Title({
+      questionData.assessment = assessment as StaffAssessmentQuestionRow['assessment'];
+      questionData.assessment_question = {
+        ...questionData.assessment_question,
+        number: selectedQuestionPosition.alternativeGroupNumber,
+        number_in_alternative_group: selectedQuestionPosition.alternativeNumber ?? null,
+      } as StaffAssessmentQuestionRow['assessment_question'];
+      questionData.alternative_group = {
+        ...questionData.alternative_group,
+        number: selectedQuestionPosition.alternativeGroupNumber,
+      } as StaffAssessmentQuestionRow['alternative_group'];
+      questionData.alternative_group_size = 1;
+      setQuestionMap((prev) => ({
+        ...prev,
+        [updatedQuestion.id!]: questionData!,
+      }));
+    }
+    // Clear validation error if we got this far
+    setQidValidationError('');
+
+    if (updatedQuestion.manualPoints !== undefined) {
+      // If we have manualPoints, we must use autoPoints (not points)
+      if (updatedQuestion.points !== undefined) {
+        updatedQuestion.autoPoints = updatedQuestion.points;
+        delete updatedQuestion.points;
+      }
+      // Also convert maxPoints to maxAutoPoints
+      if (updatedQuestion.maxPoints !== undefined) {
+        updatedQuestion.maxAutoPoints = updatedQuestion.maxPoints;
+        delete updatedQuestion.maxPoints;
       }
     } else {
-      return '—';
+      // No manual points - ensure we don't have both points and autoPoints
+      if (updatedQuestion.points !== undefined && updatedQuestion.autoPoints !== undefined) {
+        // If both exist, prefer autoPoints (since that's what we'd convert to if manualPoints were added)
+        delete updatedQuestion.points;
+        if (updatedQuestion.maxPoints !== undefined) {
+          updatedQuestion.maxAutoPoints = updatedQuestion.maxPoints;
+          delete updatedQuestion.maxPoints;
+        }
+      }
     }
-  }
 
+    const { zoneNumber, alternativeGroupNumber, alternativeNumber } = selectedQuestionPosition;
+    // Update the mappedQuestions state
+    setMappedQuestions((prevZones) => {
+      const newZones = structuredClone(prevZones); // Deep clone
+      const zone = newZones[zoneNumber - 1];
+
+      // If we're adding a new question
+      if (addQuestion) {
+        zone.questions.push({
+          ...updatedQuestion,
+          canSubmit: [],
+          canView: [],
+        });
+        return newZones;
+      }
+
+      const question = zone.questions[alternativeGroupNumber - 1];
+
+      // Check if we're editing an alternative or the alternative group itself
+      if (alternativeNumber !== undefined) {
+        // Editing an alternative question
+        if (!question.alternatives) return prevZones;
+        question.alternatives[alternativeNumber] = {
+          ...question.alternatives[alternativeNumber],
+          ...updatedQuestion,
+        };
+      } else {
+        // Editing the alternative group
```

Comment:
Nit, maybe too many comments here? If you want to keep the ones in the branches, drop the one before it.